### PR TITLE
Add `dev_id_override` to `unifi_user`

### PR DESCRIPTION
**Pull request by joshuaspence**
Allow device fingerprints to be overriden. Depends on paultyng/go-unifi#38.

### CODE REVIEW

**Comment by joshuaspence** [comment]
I'm not a huge fan of the name of the attribute... I just copied the API

File: internal/provider/resource_user_test.go [hunk 1]
```diff
@@ -187,6 +193,56 @@ func TestAccUser_existing_mac_deny(t *testing.T) {
...
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testCheckUserDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserConfig_fingerprint(testMAC, "tfacc", 123),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("unifi_user.test", "dev_id_override", "123"),
+				),
+			},
+			{
```
Comment (paultyng) [inline internal/provider/resource_user_test.go:209]: You should add the `importStep` in here to test roundtripping the information.

Comment (joshuaspence) [reply internal/provider/resource_user_test.go:209]: I missed your comment somehow. I'm not quite sure I follow, what does `importStep` do?

Comment (paultyng) [reply internal/provider/resource_user_test.go:209]: Sorry @joshuaspence forgot to respond here! `importStep` tells the test framework to drop the state, re-import the resource using just its ID, and compare the imported state to the previous state to ensure no diff (basically ensuring a clean read is the same as config+update, etc.).

Comment (joshuaspence) [reply internal/provider/resource_user_test.go:209]: Got it, that makes sense. Seems like something we should do for all test resources. I have a few weeks leave from work coming up, will likely finish this off then.

**Comment by paultyng** [comment]
Finishing this up in #166

**Pull request closed without merging**